dropwizard-http2/src/main/java/io/dropwizard/http2/Http2CConnectorFactory.java

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * Builds HTTP/2 clear text (h2c) connectors.
+ * <p/>
+ * <b>Configuration Parameters:</b>
+ * <table>
+ *     <tr>
+ *         <td>Name</td>
+ *         <td>Default</td>
+ *         <td>Description</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@code maxConcurrentStreams}</td>
+ *         <td><1024</td>
+ *         <td>
+ *             The maximum number of concurrently open streams allowed on a single HTTP/2 connection.
+ *             Larger values increase parallelism, but cost a memory commitment.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@code initialStreamSendWindow}</td>
+ *         <td>65535</td>
+ *         <td>
+ *             The initial flow control window size for a new stream. Larger values may allow greater throughput,
+ *             but also risk head of line blocking if TCP/IP flow control is triggered.
+ *         </td>
+ *     </tr>
+ * </table>
+ * <p/>
+ * For more configuration parameters, see {@link HttpsConnectorFactory}.
+ * @see HttpConnectorFactory
+ */
+@JsonTypeName("http2c")
+public class Http2CConnectorFactory extends HttpConnectorFactory {
+
+    @Min(100)
+    @Max(Integer.MAX_VALUE)
+    private int maxConcurrentStreams = 1024;
+
+    @Min(1)
+    @Max(Integer.MAX_VALUE)
+    private int initialStreamSendWindow = 65535;
+
+    @JsonProperty
+    public int getMaxConcurrentStreams() {
+        return maxConcurrentStreams;
+    }
+
+    @JsonProperty
+    public void setMaxConcurrentStreams(int maxConcurrentStreams) {
+        this.maxConcurrentStreams = maxConcurrentStreams;
+    }
+
+    @JsonProperty
+    public int getInitialStreamSendWindow() {
+        return initialStreamSendWindow;
+    }
+
+    @JsonProperty
+    public void setInitialStreamSendWindow(int initialStreamSendWindow) {
+        this.initialStreamSendWindow = initialStreamSendWindow;
+    }
+
+    @Override
+    public Connector build(Server server, MetricRegistry metrics, String name, ThreadPool threadPool) {
+
+        // Prepare connection factories for HTTP/2c
+        final HttpConfiguration httpConfig = buildHttpConfiguration();
+        final HttpConnectionFactory http11 = buildHttpConnectionFactory(httpConfig);
+        final HTTP2ServerConnectionFactory http2c = new HTTP2CServerConnectionFactory(httpConfig);
+        http2c.setMaxConcurrentStreams(maxConcurrentStreams);
+        http2c.setInitialStreamSendWindow(initialStreamSendWindow);
+
+        // The server connector should use HTTP/1.1 by default. It affords to the server to stay compatible
+        // with old clients. New clients which want to use HTTP/2, however, will make an HTTP/1.1 OPTIONS
+        // request with an Upgrade header with "h2c" value. The server supports HTTP/2 clear text connections,
+        // so it will return the predefined HTTP/2 preamble and the client and the server will switch to the
+        // new protocol.
+        return buildConnector(server, new ScheduledExecutorScheduler(), buildBufferPool(), name, threadPool,
```

Comment:
It implements `Scheduler` which in turn extends `LifeCycle`. It will be closed along with the connector.